safekeeper/src/receive_wal.rs

```diff
@@ -555,5 +592,8 @@ impl WalAcceptor {
                 next_keepalive = Instant::now() + KEEPALIVE_INTERVAL;
             }
         }
+
+        // Dropped out: timeline is shutting down
+        Ok(())
```

Comment:
Is that comment stale perhaps? WalAcceptor::run can already return Ok(()) if the self.msg_rx channel's sender is dropped.  Provisionally updating the comment in 7895178f9

Reply:
No. There is no concept of 'clean' shutdown because we never gracefully terminate the Copy protocol. The assumption is that either network futures or WalAcceptor task return some error, which may be expected (as e.g. connection reset) or not. We also want to log any disconnection with the disconnection reason.

> if the self.msg_rx channel's sender is dropped.

assumption is that some network future returns error then.

Reply:
Did this change in 248558dee8 (written after this comment thread) -- in that commit @erikgrinaker adds a flush in the code path that returns Ok() from this function -- so either the "can't happen currently" comment is now wrong, or the code in that Ok() path isn't doing anything.

Not sure why we need to distinguish Ok(()) on channel drop from cancellation though?